packages/lists/src/components/Viewport/Viewport.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { useState, useRef, useEffect } from 'react';
+import { useDebouncedTimeout } from './useDebouncedTimeout';
+import { usePreviousValue } from './usePreviousValue';
+
+export interface IViewportProps {
+  /**
+   * The height of the scroll container (in pixels).
+   */
+  height: number;
+
+  /**
+   * The width of the scroll container (in pixels).
+   */
+  width: number;
+
+  /**
+   * The child component is a function component which receives the scroll container's current scroll state as its only argument.
+   */
+  children: (viewportState: IViewportState) => JSX.Element[] | JSX.Element;
+}
+
+export type Vector2D<T> = [T, T];
+
+export enum Axis {
+  X = 0,
+  Y = 1
+}
+
+export enum ScrollDirection {
```

Comment:
Same here, what about:

```ts
export type ScrollDirection = "none" | "backward", "forward"
```

Also these could be inverted potentially so we may need to consider a different naming. Not necessary to solve in this iteration...

Reply:
I don't like to convert it to a string type since the suggested strings take more memory than an enum value (which are essentially numbers).

Reply:
Interesting.. my understanding was that enums resulted in larger bundle size due to the boilerplate required to define them and in writing their accessors compared with string union types.

https://www.typescriptlang.org/play/#code/KYOwrgtgBAIglgJ2AYwC5wPYigbwFBSFQhbAA0BRARgIbIDWA7jQgCYDOFRUAZhgszbs8AXzx5UATwAOwWIhTosAZVQI4IAOZQAvFADkJEMH1QAPgdoNBHUxf18BLWwG4gA

Would have to confirm with minified code tho.. cc: @ecraig12345 might know off-hand.

Reply:
I think the strings will end up cheaper because anything after a `.` can't be minified. You're stuck with the bytes for the value names regardless, but the strings have less additional boilerplate.
```ts
// something like this
import { ScrollDirection } from 'wherever';
const direction1 = ScrollDirection.none;
// will minify something like this, *plus* the import and enum definition boilerplate
var b=a.none;

// whereas something like this
import { ScrollDirection } from 'wherever';
const direction2: ScrollDirection = 'none';
// will minify to something like this (the import goes away)
var c='none';
```

Reply:
I was talking about memory usage. `ScrollDirection.none` is cheaper to store (since it is a numeric value `= 0`) than `none` since that is a string with four characters.

Let's keep this discussion in mind, and revisit later.

Reply:
Oh, interesting point. In this case I'm not sure whether memory usage would be the bottleneck (or whether this would contribute significantly enough to matter)? Definitely won't block the PR over it though.